ui/src/pages/Edit/index.tsx

```diff
@@ -0,0 +1,24 @@
+import * as React from "react";
+import { useParams } from "react-router-dom";
+import useSWR from "swr";
+
+import { Story } from "../../types/index";
+import { UploadStoryWrapper } from "./UploadStoryWrapper";
+
+export const Edit: React.FC = () => {
+  const { id } = useParams();
+  const { data: story, error } = useSWR<Story>(`/api/story/${id}`);
```

Comment:
Can we just make the `Edit` component have all the datafetching? We don't need a `UploadStoryWrapper` then

```ts
export const Edit: React.FC = () => {
  const { id } = useParams();
  const { data: story, error: errorStory } = useSWR<Story>(`/api/story/${id}`);
  const { data: tagOptions, error: errorTags } = useSWR<string[]>("/api/tags");
  const { data: countries, error: errorCountries } = useSWR<string[]>(
    "/api/countries"
  );

  if (errorStory) {
    return <InfoDiv>Error fetching story!</InfoDiv>;
  }
  if (errorTags) {
    return <InfoDiv>Error fetching tags!</InfoDiv>;
  }
  if (errorCountries) {
    return <InfoDiv>Error fetching countries array!</InfoDiv>;
  }

  if (tagOptions === undefined || countries === undefined || story === undefined) {
    return <StyledProgress />;
  }

  return (
    <UploadStory
      id={id}
      currentStory={story}
      bio={story.author.bio}
      tagOptions={tagOptions}
      countries={countries}
    />
  );

}
```

Reply:
I feel like that would be more redundant because we would have to copy the same code over to the Upload component as well. Having the wrapper allows us to fetch the data common to both pages via one component. If you think the way you suggested is still more efficient, pls let me know and I can make the change.